internal/sys/password_policy.go

```diff
@@ -0,0 +1,252 @@
+package sys
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/base"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/client"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func NewPasswordPolicyResource() resource.Resource {
+	return &PasswordPolicyResource{}
+}
+
+type PasswordPolicyResource struct {
+	meta *provider.ProviderMeta
+}
+
+func (r *PasswordPolicyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_password_policy_fw"
+}
+
+// PasswordPolicyModel describes the Terraform resource data model to match the
+// resource schema.
+type PasswordPolicyModel struct {
+	Namespace types.String `tfsdk:"namespace"`
+
+	Name   types.String `tfsdk:"name"`
+	Policy types.String `tfsdk:"policy"`
+}
+
+// PasswordPolicyResourceAPIModel describes the Vault API data model.
+type PasswordPolicyResourceAPIModel struct {
+	Policy string `json:"policy" mapstructure:"policy"`
+}
+
+func (r *PasswordPolicyResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	meta, ok := req.ProviderData.(*provider.ProviderMeta)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *provider.ProviderMeta, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.meta = meta
+}
+
+func (r *PasswordPolicyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the password policy.",
+				Required:            true,
+			},
+			"policy": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The password policy document",
+			},
+		},
+		MarkdownDescription: "Provides a resource to manage Password Policies.",
+	}
+
+	base.MustAddBaseSchema(&resp.Schema)
+}
+
+func (r *PasswordPolicyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan *PasswordPolicyModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
```

Comment:
Wondering if this should be `var plan PasswordPolicyModel` instead of a `*PasswordPolicyModel`? Looks like we're passing a pointer to the pointer on L80. Might work okay but noticed they do something different in https://developer.hashicorp.com/terraform/tutorials/providers-plugin-framework/providers-plugin-framework-resource-create#implement-create-functionality.

Also relevant in other resource CRUD methods below.

Reply:
hmm, yeah I was following along with https://developer.hashicorp.com/terraform/plugin/framework/resources/create but the double pointer shouldn't be necessary.